src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -606,8 +616,15 @@
         (inc-restart-counter! this)
         this)
       (a/stop [this]
-        (shutdown! app-context)
-        this)
+        (a/stop this false))
+      (a/stop [this throw?]
+        (if-let [errors (not-empty (shutdown! app-context))]
+          (let [msg (i18n/trs "Error during app shutdown!")
+                e (ex-info msg {:errors errors})]
+            (log/error e msg)
+            (when throw?
+              (throw e)))
```

Comment:
Possible regression: Perhaps `(if thrown? (throw e) this)` to avoid breaking changes.

Reply:
thanks!